core/change.go

```diff
@@ -100,79 +105,149 @@ func (change Change) Write(writer io.Writer) error {
 	return err
 }
 
-func kindFromLabel(config Config, label string) *KindConfig {
-	for _, kindConfig := range config.Kinds {
-		if kindConfig.Label == label {
-			return &kindConfig
-		}
-	}
-
-	panic("label not part of any kind")
+type PromptContext struct {
+	config      Config
+	stdinReader io.ReadCloser
+	kind        *KindConfig
 }
 
 // AskPrompts will ask the user prompts based on the configuration
 // updating the change as prompts are answered.
-func AskPrompts(change *Change, config Config, stdinReader io.ReadCloser) error {
-	var (
-		err  error
-		kind *KindConfig
-	)
-
-	if len(config.Components) > 0 {
-		compPrompt := promptui.Select{
-			Label: "Component",
-			Items: config.Components,
-			Stdin: stdinReader,
-		}
+func (change *Change) AskPrompts(config Config, stdinReader io.ReadCloser) error {
+	ctx := PromptContext{
+		config:      config,
+		stdinReader: stdinReader,
+		kind:        nil,
+	}
 
-		_, change.Component, err = compPrompt.Run()
-		if err != nil {
-			return err
-		}
+	err := change.promptForComponent(&ctx)
+
+	if err != nil {
+		return err
 	}
 
-	if len(config.Kinds) > 0 {
-		kindPrompt := promptui.Select{
-			Label: "Kind",
-			Items: config.Kinds,
-			Stdin: stdinReader,
-		}
+	err = change.promptForKind(&ctx)
 
-		_, change.Kind, err = kindPrompt.Run()
-		if err != nil {
-			return err
+	if err != nil {
+		return err
+	}
+
+	err = change.parseKind(&ctx)
+
+	if err != nil {
+		return err
+	}
+
+	err = change.promptForBody(&ctx)
+
+	if err != nil {
+		return err
+	}
+
+	return change.promptForUserChoices(&ctx)
+}
+
+func (change *Change) promptForComponent(ctx *PromptContext) error {
+	if len(ctx.config.Components) == 0 {
+		return nil
+	}
+
+	compPrompt := promptui.Select{
+		Label: "Component",
+		Items: ctx.config.Components,
+		Stdin: ctx.stdinReader,
+	}
+
+	var err error
+	_, change.Component, err = compPrompt.Run()
+
+	return err
+}
+
+func (change *Change) promptForKind(ctx *PromptContext) error {
+	if len(ctx.config.Kinds) == 0 || len(change.Kind) > 0 {
+		return nil
+	}
+
+	kindPrompt := promptui.Select{
+		Label: "Kind",
+		Items: ctx.config.Kinds,
+		Stdin: ctx.stdinReader,
+	}
+
+	var err error
+	_, change.Kind, err = kindPrompt.Run()
+
+	return err
+}
+
+func (change *Change) parseKind(ctx *PromptContext) error {
+	if len(change.Kind) == 0 {
+		return nil
+	}
+
+	for i := range ctx.config.Kinds {
+		kindConfig := &ctx.config.Kinds[i]
+
+		if kindConfig.Label == change.Kind {
+			ctx.kind = kindConfig
+			return nil
 		}
+	}
+
+	return fmt.Errorf("%w: %s", errInvalidKind, change.Kind)
+}
 
-		kind = kindFromLabel(config, change.Kind)
+func (change *Change) promptForBody(ctx *PromptContext) error {
+	if change.expectsNoBody(ctx) && len(change.Body) > 0 {
+		return errKindDoesNotAcceptBody
```

Comment:
```suggestion
		return fmt.Errorf("%w: %s", errKindDoesNotAcceptBody, change.Kind)
```
I would include the kind here as well